config/config.go

```diff
@@ -176,3 +180,14 @@ func (c *Config) ConfigurePlugins(ctx context.Context, path string) (map[string]
 	}
 	return sources, destinations, nil
 }
+
+func firstPath(paths []string, filename string) (string, error) {
+	err := os.ErrNotExist
```

Comment:
This error is _only_ ever returned IF AND ONLY IF no paths are given, right? Otherwise, we will either return the first `fpath` or the error from trying to stat the final `fpath`, right?

Reply:
yes

Reply:
I think that the semantics that would make more sense to me would be:
1. try to stat a path
2. if no error, then return the file
3. if there was an error, then
    1. check if the error was `os.ErrNotExist`
    2. if so, then skip and continue the loop
    3. if not, then return the unexpected error
4. if nothing was found after the last path, then return `os.ErrNotExist`

Essentially, we treat `os.ErrNotExist` as expected errors and all other errors as failures that should be returned immediately. WDYT?

Reply:
so in case we don't have read permission for the first path, we would exit without trying the next one?
The Stat operations would return the error type PathError and not ErrNotExist (it can be wrapped inside though)
```go	
if err != nil {
	return nil, &PathError{Op: "stat", Path: name, Err: err}
}
```
The `err` is whatever the syscall returns. So it could be something other then `ErrNotExist`. I think realistically it could be `ErrPermission`, too.

Reply:
Yes, exactly, we'd want to check the error i think